reactive/kubernetes_control_plane.py

```diff
@@ -3704,6 +3704,19 @@ def request_ceph_permissions():
     set_flag("kubernetes-control-plane.ceph.permissions.requested")
 
 
+@when_any("config.changed.image-registry", "cni.available")
+def image_registry_changed():
+    registry = hookenv.config("image-registry")
+    # Share image-registry with cni requirers
+    cni = endpoint_from_flag("cni.available")
+    if cni:
+        endpoint_from_flag("cni.available").set_image_registry(registry)
+    else:
+        hookenv.log(
+            "CNI endpoint not available yet, waiting to set image registry data"
+        )
```

Comment:
so, if you configure this charm before the CNI interface is available, then the `image-registry` won't be set in the CNI relation.  

You should consider setting it during the `@when("cni.available")` hook as well

Reply:
```
when_any("config.changed.image-registry", "cni.available")
```
is that not enough for that or would it need to be a separate hook? if the config changes before cni is available, it will log it, then when cni is available it should set the data

Reply:
OH OH OH!  I missed the `when_any`